profile.go

```diff
@@ -0,0 +1,199 @@
+// Copyright 2021 Contributors to the Veraison project.
+// SPDX-License-Identifier: Apache-2.0
+
+package eat
+
+import (
+	"encoding/asn1"
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"strconv"
+	"strings"
+)
+
+// Profile is either a well formed RFC3986 compliant URI(uri  *url.URL).
+// or an ObjectIdentifier  ([]int)
+type Profile struct {
+	val interface{}
+}
+
+// NewProfile instantiates a new Profile object from a given input string
+// The input string can either be a well formed URI as string
+// OR it could be a OID formated as a string example "1.2.3.4"
+func NewProfile(urlOrOID string) (*Profile, error) {
+	p := Profile{}
+	if err := p.SetProfile(urlOrOID); err != nil {
+		return nil, err
+	}
+	return &p, nil
+}
+
+// SetProfile sets the given urlOrOID overwriting a previously stored
+// value
+func (s *Profile) SetProfile(urlOrOID string) error {
+	// First attempt to decode input string as a URL
+	u, err := url.Parse(urlOrOID)
+	if err != nil || !u.IsAbs() {
+		s.val, err = decodeOIDfromString(urlOrOID)
```

Comment:
this would overwrite the internal value even on error, which is not what we want
```suggestion
		v, err = decodeOIDfromString(urlOrOID)
```